src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +111,1836 @@ const ActorInit Boss_Goma_InitVars = {
+        case 32: // eyelid (bottom)
+            rot->x += this->eyeLidBottomRotX;
+            break;
+        case 35: // eyelid (top)
+            rot->x += this->eyeLidTopRotX;
+            break;
+        case 38: // eye (iris) (parent of 39)
+            rot->x += this->eyeIrisRotX;
+            rot->y += this->eyeIrisRotY;
+            break;
+        case 39: // eye (iris)
+            if (this->eyeState == EYESTATE_IRIS_FOLLOW_BONUS_IFRAMES && this->eyeLidBottomRotX < -0xA8C) {
+                *dList = NULL;
+            } else {
+                if (this->visualState == VISUALSTATE_DEFEATED) {
+                    gDPSetEnvColor(oGfxCtx->polyOpa.p++, 50, 50, 50, 255);
+                } else {
+                    gDPSetEnvColor(oGfxCtx->polyOpa.p++, 255, 255, 255, 255);
+                }
+                Matrix_JointPosition(pos, rot);
+                if (*dList != NULL) {
+                    Matrix_Push();
+                    Matrix_Scale(this->eyeIrisScaleX, this->eyeIrisScaleY, 1.0f, MTXMODE_APPLY);
+                    gSPMatrix(oGfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_goma.c", 4815),
+                              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                    gSPDisplayList(oGfxCtx->polyOpa.p++, *dList);
+                    Matrix_Pull();
+                }
+                doNotDrawLimb = true;
+            }
+            break;
+        // tail parts
+        case 11:
+        case 12:
+        case 13:
+        case 14:
+            Matrix_JointPosition(pos, rot);
+            if (*dList != NULL) {
+                Matrix_Push();
+                Matrix_Scale(this->tailLimbsScale[limbIndex - 11], this->tailLimbsScale[limbIndex - 11],
+                             this->tailLimbsScale[limbIndex - 11], MTXMODE_APPLY);
+                gSPMatrix(oGfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_goma.c", 4836),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(oGfxCtx->polyOpa.p++, *dList);
+                Matrix_Pull();
+            }
+            doNotDrawLimb = true;
+            break;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_80919150.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_goma.c", 4858);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_80919278.s")
+    return doNotDrawLimb;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091932C.s")
+void BossGoma_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f tailZero = { 0, 0, 0 };
+    static Vec3f clawBackLocalPos = { 0, 0, 0 };
+    static Vec3f focusEyeLocalPos = { 0, 300, 2650 }; // in the center of the surface of the lens
+    static Vec3f zero = { 0, 0, 0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_809193EC.s")
+    Vec3f childPos;
```

Comment:
no need to put a line break between static and stack vars, they shouldn't be treated as a different list.

make sure to put .0f on the values in your vectors, might be relevant throughout the file